ur'strings' are a Syntax Errors in Python 3

In Python 3, you can have __u'strings'__ and you can have __r'strings'__ but you can not have __ur'strings'__.
* __python3 -c "print(ur'string')"__  # --> Syntax Error

ur'strings' currently appear in three files...

$ __python3 -m flake8 . --count --select=E901,E999 --show-source --statistics__
```
./ckan/tests/test_coding_standards.py:127:55: E999 SyntaxError: invalid syntax
    pattern = re.compile(ur'#.*?coding[:=][ \t]*utf-?8')
                                                      ^
./ckan/lib/munge.py:159:42: E999 SyntaxError: invalid syntax
    filename = re.sub(ur'[^a-zA-Z0-9_. -]', '', filename).replace(u' ', u'-')
                                         ^
./doc/conf.py:341:49: E999 SyntaxError: invalid syntax
  ('contents', 'CKAN.tex', ur'CKAN documentation',
                                                ^
3     E999 SyntaxError: invalid syntax
```

The last one is governed by:
* http://www.sphinx-doc.org/en/stable/config.html#confval-latex_documents

Based on the [example __conf.py__ file](http://www.sphinx-doc.org/en/stable/config.html#example-of-configuration-file), I believe that we should use u'strings' in this context.  #4049

--- a/ckan/lib/munge.py
+++ b/ckan/lib/munge.py
@@ -158,8 +158,8 @@ def munge_filename(filename):
     # Clean up
     filename = filename.lower().strip()
     filename = substitute_ascii_equivalents(filename)
-    filename = re.sub(ur'[^a-zA-Z0-9_. -]', '', filename).replace(u' ', u'-')
-    filename = re.sub(ur'-+', u'-', filename)
+    filename = re.sub(u'[^a-zA-Z0-9_. -]', '', filename).replace(u' ', u'-')
+    filename = re.sub(u'-+', u'-', filename)
 
     # Enforce length constraints
     name, ext = os.path.splitext(filename)

--- a/ckan/tests/test_coding_standards.py
+++ b/ckan/tests/test_coding_standards.py
@@ -18,6 +18,7 @@ import subprocess
 import sys
 
 from six import text_type
+from six.moves import xrange
 
 FILESYSTEM_ENCODING = text_type(
     sys.getfilesystemencoding() or sys.getdefaultencoding()
@@ -125,7 +126,7 @@ def test_source_files_specify_encoding():
 
     Empty files and files that only contain comments are ignored.
     '''
-    pattern = re.compile(ur'#.*?coding[:=][ \t]*utf-?8')
+    pattern = re.compile(u'#.*?coding[:=][ \\t]*utf-?8')
     decode_errors = []
     no_specification = []
     for abs_path, rel_path in walk_python_files():

